lib/THC/THCTensorRandom.cuh

```diff
@@ -37,6 +37,69 @@ __global__ void generateLogNormal<double>(curandStateMtgp32 *state, int size, do
   }
 }
 
+
+template <typename T>
+__global__ void
+multinomialAliasDrawKernel(curandStateMtgp32* state, int size, long *output, long *J, T *q, long K,  T *uniform, T *bernoulli)
+{
+  long idx = blockIdx.x * BLOCK_SIZE + threadIdx.x;
+       
+  if (idx < size)
+    {
+      long rand_ind = ScalarConvert<T, long>::to(uniform[idx]);
+      
+       T bern_uniform = bernoulli[idx];
+       int _mask = ScalarConvert<bool, int >::to( THCNumerics<T>::lt(bern_uniform, q[rand_ind]));
```

Comment:
`ScalarConvert<bool, int>` is unnecessary, it's just a static cast of `int` to `bool`

`ScalarConvert` is mostly meant for types for which the constructor or casting is non-obvious; e.g., float or int -> half